src/RemoteInboxNotifications/ProductsProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Provider for product-related queries and operations.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Provider for product-related queries and operations.
+ */
+class ProductsProvider {
+	/**
+	 * Returns whether or not there are any products.
+	 *
+	 * @return boolean If there are any products.
+	 */
+	public function are_there_products() {
+		return $this->get_product_count() > 0;
+	}
+
+	/**
+	 * Returns the number of products.
+	 *
+	 * @return integer The number of products.
+	 */
+	public function get_product_count() {
```

Comment:
Should we use [getProducts()](https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-product-query.php#L74) here?

Reply:
Thanks 🙏 done. I also added some unit tests since a little more logic ended up in the processor.